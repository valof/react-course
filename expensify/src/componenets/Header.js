import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = () => (
<header>
    <h1>Expensify!</h1>
    <NavLink to="/" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} >Go Home!</NavLink>
    <br></br>
    <NavLink to="/create" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} >Go Create!</NavLink>
    <br></br>
    <NavLink to="/help" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} >Help...</NavLink>
</header>
);
  
  export default Header;
  
