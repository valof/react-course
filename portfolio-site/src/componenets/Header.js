import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = () => (
<header>
    <h1>Portfolio!</h1>
    <NavLink to="/" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} exact={"true"}>Go Home!</NavLink>
    <NavLink to="/portfolio" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} exact={"true"}>Portfolio!</NavLink>
    <NavLink to="/contact" className={({ isActive }) => (isActive ? 'is-active' : 'inactive')} >Contact</NavLink>
</header>
);
  
  export default Header;
  
