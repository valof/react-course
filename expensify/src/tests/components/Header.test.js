import React from 'react'
import { MemoryRouter } from "react-router-dom";
import TestRenderer from 'react-test-renderer';
import Header from '../../componenets/Header'

test ('should render Header say "Expensify!" and render correctly', () => {
    const testRenderer = TestRenderer.create(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
      );
      const componentInstance = testRenderer.root;
      expect(componentInstance.findByType('h1').props.children).toBe('Expensify!')
      expect(testRenderer.toJSON()).toMatchSnapshot();
})




















    //const mountedHeader = create(<Header />)
    // const renderer = ReactShallowRenderer.createRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    // console.log(renderer.getRenderOutput())
