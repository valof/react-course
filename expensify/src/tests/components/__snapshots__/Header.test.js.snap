// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header say "Expensify!" and render correctly 1`] = `
<header>
  <h1>
    Expensify!
  </h1>
  <a
    aria-current="page"
    className="is-active"
    href="/"
    onClick={[Function]}
  >
    Go Home!
  </a>
  <br />
  <a
    className="inactive"
    href="/create"
    onClick={[Function]}
  >
    Go Create!
  </a>
  <br />
  <a
    className="inactive"
    href="/help"
    onClick={[Function]}
  >
    Help...
  </a>
</header>
`;
